src/Raven.Client/ServerWide/Tcp/TcpNegotiation.cs

```diff
@@ -25,16 +25,50 @@ public static int NegotiateProtocolVersion(JsonOperationContext documentsContext
                     });
                     writer.Flush();
                     var version = parameters.ReadRespondAndGetVersion(documentsContext, writer);
+                    //In this case we usally throw internaly but for completeness we better handle it
+                    if (version == -2)
+                    {
+                        return new TcpFeaturesSupported(parameters.Operation, -2);
+                    }
                     var (supported, prevSupported) = TcpConnectionHeaderMessage.OperationVersionSupported(parameters.Operation, version);
                     if (supported)
-                        return version;
+                        return new TcpFeaturesSupported(parameters.Operation, version);
                     if (prevSupported == -1)
-                        return -1;
+                        return new TcpFeaturesSupported(parameters.Operation, -1);
                     currentVersion = prevSupported;
                 }
             }
         }
     }
+
+    public class TcpFeaturesSupported
+    {
+        private int _version;
+        private TcpConnectionHeaderMessage.OperationTypes _type;
+
+        public TcpFeaturesSupported(TcpConnectionHeaderMessage.OperationTypes type, int protocolVersion)
+        {
+            _version = protocolVersion;
+            _type = type;
+        }
+
+        public bool IsMissingAttachmentSupported {
```

Comment:
Wrap it inside some class that can be access via 'Replication' property. It would be more clear what bool is for what feature. e.g. `features.Replication.SupportsMissingAttachments` ?